src/pages/VideoCall/index.js

```diff
@@ -0,0 +1,108 @@
+import React, { useEffect, useRef, useState } from 'react'
+import styled from 'styled-components'
+import {
+    useParams
+} from 'react-router-dom'
+import DailyIframe from '@daily-co/daily-js'
+
+import Header from '../../components/Header'
+import Footer from '../../components/Footer'
+
+const IframeStyled = styled.div`
+    width: 100vw;
+    height: 81vh;
+    color: ${({theme}) => theme.color.white};
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    font-size: ${({theme}) => theme.font.L};
+    iframe {
+        width: 100%;
+        height: 100%;
+        border: none;   
+    }
+`
+const LeaveButton = styled.div`
+    width: 100vw;
+    height: 7vh;
+    display: flex;
+    align-items: center;
+    color: ${({theme}) => theme.color.white};
+    font-weight: 600;
+    background-color: ${({theme}) => theme.color.red};
+    cursor: pointer;
+    :hover {
+        background-color: #9a2530;
+    }
+    p {
+        margin: 0 auto;
+    }
+`
+
+const VideoCallFrame = () => {
+    const [ leftCallFrame, setLeftCallFrame ] = useState(false)
+    let { videoName } = useParams()
+
+    const url = `https://instantvisio.daily.co/${videoName}`
+
+    const videoFrame = useRef(null)
+
+    // to display confirmation message 
+    // when user attempts leaving page
+    const leavingCallPage = (event) => {
+        // Cancel the event as stated by the standard.
+        event.preventDefault()
+        // Chrome requires returnValue to be set.
+        event.returnValue = ''
+    }
+
+    const leavingCallFrame = () => {
+        setLeftCallFrame(true)
+    }
+
+    useEffect(() => {
+        const daily = DailyIframe.wrap(videoFrame.current)
+        
+        daily.join({
+            url,
+            showFullscreenButton: true
+        })
+
+        if (leftCallFrame) {
+            daily.leave()
+        }
+
+        window.addEventListener('beforeunload', leavingCallPage)
+        // ComponentWillUnmount
+        return () => {
+            window.removeEventListener('beforeunload', leavingCallPage)
+        }
+       
+    }, [])
```

Comment:
better

```suggestion
    }, [url, leftCallFrame])
```

Reply:
added url.

had previously added "leftCallFrame" but the attached error is generated

![leftCallFrame_error](https://user-images.githubusercontent.com/50984435/78251717-069bc180-74f2-11ea-87bd-efa37a6bc374.png)

Reply:
very weird, i try with these ref and i didn't have the error